src/Nethermind/Nethermind.Evm.Test/Eip1344Tests.cs

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2018 Demerzel Solutions Limited
+ * This file is part of the Nethermind library.
+ *
+ * The Nethermind library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Nethermind library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+using Nethermind.Core.Extensions;
+using NUnit.Framework;
+
+namespace Nethermind.Evm.Test
+{
+    [TestFixture]
+    public class Eip1344Tests : VirtualMachineTestsBase
+    {
+        [Test]
+        public void chain_id_opcode_puts_expected_value_onto_the_stack()
+        {
+            byte[] code = Prepare.EvmCode
+                .Op(Instruction.CHAINID)
+                .PushData(0)
+                .Op(Instruction.SSTORE)
+                .Done;
+            
+            var result = Execute(code);
+            Assert.AreEqual(StatusCode.Success, result.StatusCode);
+            AssertGas(result, 21000 + GasCostOf.VeryLow + GasCostOf.Base + GasCostOf.SSet);
+            AssertStorage(0, SpecProvider.ChainId.ToBigEndianByteArray());
```

Comment:
not a great test if tested with 0 - can we test with different chain IDs? 0, 1, 32000?